### PR TITLE
fix: disabled long press menu when viewing some other channel

### PR DESCRIPTION
**Pull request by chungwwei**

### CODE REVIEW

File: lib/components/chat_history/twitch/message_link_preview.dart [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import 'package:metadata_fetch/metadata_fetch.dart';
 import 'package:flutter/material.dart';
 
+Map<String, _TwitchClipData> linkPreviewCache = {};
```
Comment (kevmo314) [inline lib/components/chat_history/twitch/message_link_preview.dart:4]: Can you remove the link preview change from this PR?

File: lib/components/chat_history/message.dart [hunk 2]
```diff
@@ -28,6 +28,14 @@ class ChatHistoryMessage extends StatelessWidget {
         if (layoutModel.isInteractionLockable && layoutModel.locked) {
           return child;
         }
+        final userModel = Provider.of<UserModel>(context, listen: false);
+        final loginChannel = userModel.userChannel!.displayName;
```
Comment (kevmo314) [inline lib/components/chat_history/message.dart:32]: Can we compare by channel id? The format for `TwitchMessageModel.channelId` is `$provider:$channelId`. This is more robust when we add other providers.

File: lib/components/chat_history/twitch/message_link_preview.dart (hunk 1 above)
Comment (chungwwei) [reply lib/components/chat_history/twitch/message_link_preview.dart]: I copied and pasted the file `message_link_preview.dart` from main to my local one, and pushed it.

File: lib/components/chat_history/message.dart (hunk 2 above)
Comment (chungwwei) [reply lib/components/chat_history/message.dart]: I added `channelId` to `TwitchMessageModel`, and I changed it so its compared by `channelId`.

**Review by kevmo314 — APPROVED** [review]

**Pull request merged**